docs/src/developer/gpu_computations.md

```diff
@@ -21,33 +21,64 @@ when creating the basis. `architecture` should be an initialized instance of
 the (non-exported) `CPU` and `GPU` structures. `CPU` does not require any argument,
 but `GPU` requires the type of array which will be used for GPU computations.
 
-```@example gpu_computations
+```julia
 PlaneWaveBasis(model; Ecut, kgrid, architecture = DFTK.CPU())
 PlaneWaveBasis(model; Ecut, kgrid, architecture = DFTK.GPU(CuArray))
 ```
-Those `CPU` and `GPU` structures we inspired by the structures of same name in
-[Oceananigans](https://github.com/CliMA/Oceananigans.jl). **However, it is very**
-**likely that this API will change.** We hope that a unified convention will emerge,
-instead of having every package define their own way of dealing with computations on
-multiple architectures.
+**It is very likely that this API will change.**, based on the evolution of the
+Julia ecosystem concerning distributed architectures.
 
 Not all terms can be used when doing GPU computations: currently, the `Anyonic`,
-`Magnetic`, `TermPairwisePotential` and `Xc` terms are not supported. Symmetries
-should also be disabled. Nothing has yet been made to enable GPU support for
-automatic differentiation.
+`Magnetic`, `TermPairwisePotential` and `Xc` terms are not supported. Some mixings,
+such as `χ0Mixing` are also not yet supported. GPU features are not yet exhaustively
+ tested, and it is likely that some aspects of the code such as automatic
+ differentiation or stresses will not work.
 
 ## Pitfalls
 There are a few things to keep in mind when doing GPU programming in DFTK.
-- Transfers to and from a device are a bit tricky because they are done by
-converting the array to an other type (ex: `CuArray(A)`, `Array(A)`...). This
-means that the array type used for computations should be stored somewhere:
-currently, this is done by the `CPU` and `GPU` structures and the helper functions
-`to_device` and `to_cpu`. `similar` could also be used, but then a reference array
+- Transfers to and from a device can be done simply by converting an array to
+an other type. However, hard-coding the new array type (such as writing
+`CuArray(A)` to move `A` to a CUDA GPU) is not cross-architecture, and can
+be confusing for developers working only on the CPU code. These data transfers
+should be done using the helper functions `to_device` and `to_cpu` which
+provide a level of abstraction while also allowing multiple architectures
+to be used.
+```julia
+cuda_gpu = DFTK.GPU(CuArray)
+cpu_architecture = DFTK.CPU()
+A = rand(10) # A is on the CPU
+B = DFTK.to_device(cuda_gpu, A) # B is a copy of A on the CUDA GPU
+B .+= 1.
+C = DFTK.to_cpu(B) # C is a copy of B on the CPU
+D = DFTK.to_device(cpu_architecture, B) # Equivalent to the previous line, but      should be avoided as it is less clear
+```
+*Note:* `similar` could also be used, but then a reference array
 (one which already lives on the device) needs to be available at any time.
```

Comment:
at any time. Hmm. Rather at call time (similar allocates a new array).

Reply:
I meant that when you are developing code, you always need to have some sort of reference array at hand: sometimes, this means adding a new argument to a function which will only be used to know what type of device is being used.

Reply:
Well but in this case you need the `cpu_architecture` at hand. I don't see why one is an advantage over the other.

Reply:
Hum, that's true.